src/payment/strategies/chasepay/chasepay-payment-strategy.ts

```diff
@@ -0,0 +1,224 @@
+import { RequestSender, Response } from '@bigcommerce/request-sender';
+
+import {
+    PaymentActionCreator,
+    PaymentInitializeOptions,
+    PaymentMethod,
+    PaymentMethodActionCreator,
+    PaymentRequestOptions,
+    PaymentStrategyActionCreator,
+} from '../..';
+import { CheckoutActionCreator, CheckoutStore, InternalCheckoutSelectors } from '../../../checkout';
+import { InvalidArgumentError, MissingDataError, MissingDataErrorType } from '../../../common/error/errors';
+import { toFormUrlEncoded } from '../../../common/http-request';
+import { OrderActionCreator, OrderRequestBody } from '../../../order';
+import { ChasePayScriptLoader, ChasePaySuccessPayload } from '../../../payment/strategies/chasepay';
+import { PaymentArgumentInvalidError } from '../../errors';
+import { CryptogramInstrument } from '../../payment';
+import PaymentStrategy from '../payment-strategy';
+import WepayRiskClient from '../wepay/wepay-risk-client';
+
+export default class ChasepayPaymentStrategy extends PaymentStrategy {
+    private _paymentMethod?: PaymentMethod;
+    private _logoContainer?: string;
+
+    constructor(
+        store: CheckoutStore,
+        private _paymentMethodActionCreator: PaymentMethodActionCreator,
+        private _chasePayScriptLoader: ChasePayScriptLoader,
+        private _paymentActionCreator: PaymentActionCreator,
+        private _orderActionCreator: OrderActionCreator,
+        private _requestSender: RequestSender,
+        private _wepayRiskClient: WepayRiskClient,
+        private _paymentStrategyActionCreator: PaymentStrategyActionCreator,
+        private _checkoutActionCreator: CheckoutActionCreator
+
+    ) {
+        super(store);
+    }
+
+    initialize(options: PaymentInitializeOptions): Promise<InternalCheckoutSelectors> {
+        const { methodId } = options;
+
+        if (!options.chasepay) {
+            throw new InvalidArgumentError('Unable to initialize payment because "options.chasepayCheckoutOptions" argument is not provided.');
+        }
+
+        const {
+            onPaymentSelect = () => {},
+            logoContainer,
+            chasePayActionButton,
+        } = options.chasepay;
+
+        this._logoContainer = logoContainer;
+
+        return this._store.dispatch(this._paymentMethodActionCreator.loadPaymentMethod(methodId))
```

Comment:
Why do you need to make a call to load the payment method here? As far I know, you only need to load it in order to get the `digitalSessionId` for launching the ChasePay lightbox, which you don't do here.

Reply:
We need the `digitalSessionId` and the `config.testMode` to load the ChasepayLightbox, so we require to call the `paymentMethod` or is there any other method to do it?

Reply:
I understand you need to call `loadPaymentMethod` just before you launch the ChasePay lightbox - inside the `click` listener - because you need the `digitalSessionId`. But not here. `config.testMode` should already be available so you don't need to make an additional call.

Also, you should retrieve the `digitalSessionId` inside the `click` handler because there's a chance that the session ID might be expired by the time `click` event gets triggered.

Reply:
Ok, got it.